cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1444,6 +1467,41 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
             return Future.succeededFuture(this);
         }
 
+        Future<ReconciliationState> kafkaPersistentClaimDeletion() {
+            if (kafkaCluster.getStorage() instanceof PersistentClaimStorage) {
+
+                PersistentClaimStorage storage = (PersistentClaimStorage) kafkaCluster.getStorage();
+                for (int i = 0; i < kafkaCluster.getReplicas(); i++) {
+
+                    PersistentVolumeClaim pvc = annotateDeleteClaim(reconciliation.namespace(),
+                            AbstractModel.VOLUME_NAME + "-0-" + KafkaCluster.kafkaClusterName(reconciliation.name()) + "-" + i,
+                            storage.isDeleteClaim());
+                    if (pvc != null) {
+                        pvcOperations.reconcile(namespace, pvc.getMetadata().getName(), pvc);
+                    }
+                }
+            } else if (kafkaCluster.getStorage() instanceof JbodStorage) {
+
+                JbodStorage storage = (JbodStorage) kafkaCluster.getStorage();
+                for (int i = 0; i < kafkaCluster.getReplicas(); i++) {
+
+                    for (SingleVolumeStorage volume : storage.getVolumes()) {
+
+                        if (volume instanceof PersistentClaimStorage) {
+
+                            PersistentVolumeClaim pvc = annotateDeleteClaim(reconciliation.namespace(),
```

Comment:
What do you think the empty lines above are adding? I understand the use of empty lines to emphasize lumps of code doing different things, but the empty lines above just seem to be unhelpful to me.